tasks/build/download_node_builds.js

```diff
@@ -1,27 +1,95 @@
 module.exports = function (grunt) {
   let { map, fromNode } = require('bluebird');
-  let { resolve } = require('path');
-  let { pluck } = require('lodash');
-  let { createWriteStream } = require('fs');
+  let { resolve, basename } = require('path');
+  let { createReadStream, createWriteStream } = require('fs');
   let { createGunzip } = require('zlib');
   let { Extract } = require('tar');
   let { rename } = require('fs');
+  let checkHash  = require('check-hash');
   let wreck = require('wreck');
 
-  let platforms = grunt.config.get('platforms');
-  let activeDownloads = [];
+  const platforms = grunt.config.get('platforms');
+  let shaSums = {};
+  const getShaSums = async (done) => {
+    const nodeVersion = grunt.config.get('nodeVersion');
+    const shaSumsUri = `https://nodejs.org/dist/v${nodeVersion}/SHASUMS256.txt`;
 
-  let start = async (platform) => {
-    let finalDir = platform.nodeDir;
-    let downloadDir = `${finalDir}.temp`;
+    const resp = await fromNode(cb => {
```

Comment:
The potential errors thrown by this `await` need to be caught and handled (logged with grunt.log, and if fatal, kill the execution with grunt.fail), right now they're just getting gobbled up. Same goes for all the other `await`s . This could possibly be handled in a generic way in the task registration function by adding a .catch() to the returned promise.